trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1227,7 +1265,40 @@ public Response createDataProvider(
                 @RequestBody(description = CFG_CREATE_DESC + " " + CFG_KEYS_DESC, content = {
                         @Content(schema = @Schema(implementation = org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.ConfigurationQueryParameters.class))
                 }, required = true) ConfigurationQueryParameters queryParameters) {
-        return Response.status(Status.NOT_IMPLEMENTED).entity("POST a configuration to a given output to create derived output").build(); //$NON-NLS-1$
+
+        try (FlowScopeLog scope = new FlowScopeLogBuilder(LOGGER, Level.FINE, "DataProviderService#createDataProvider") //$NON-NLS-1$
+                .setCategory(outputId).build()) {
+            TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+            if (experiment == null) {
+                return Response.status(Status.NOT_FOUND).entity(NO_SUCH_TRACE).build();
+            }
+
+            Response errorResponse = validateOutputConfigParameters(outputId, queryParameters);
+            if (errorResponse != null) {
+                return errorResponse;
+            }
+
+            IDataProviderFactory factory = manager.getFactory(outputId);
+            if (factory == null) {
+                return Response.status(Status.METHOD_NOT_ALLOWED).entity(NO_SUCH_PROVIDER).build();
```

Comment:
It seems this status should only be used when the method (e.g. POST, DELETE) is not allowed for this endpoint.

Reply:
Makes sense. I'll update what is returned and align the API documentation